manifests/kustomize/base/installs/multi-user/pipelines-profile-controller/sync.py

```diff
@@ -16,6 +16,9 @@
 import json
 import os
 import base64
+import boto3, botocore, secrets, string
```

Comment:
We prefer not to include cloud provider associated library for the default installation path. boto3 and botocore are from AWS, can we remove them from here?

Reply:
Yes, i could make a runtime check and install them if they are not already there (self-built) images

Reply:
Why not just minion python client only? I'm not familiar with the aws packages here, but it looks like your code would only work for S3 while I don't see equivalent implementation for GCS.

Reply:
> Why not just minion python client only? I'm not familiar with the aws packages here, but it looks like your code would only work for S3 while I don't see equivalent implementation for GCS.

Because sadly the python client uses the mc binary for some stuff. Yes it only works for minio S3 at the moment. @zijianjoy had some suggestions to use overlays for different Providers (Minio, AWS, GCS). I asked this almost 40 days ago in this PR and mentioned it explicitly in the KFP meeting, but i still have no reply. Can you comment on the structure question below?